### ISSUE TITLE
Include Core Version in Module Metadata section

### ISSUE THREAD

**Issue by rainman110** [dlr-gtlab/gtlab-core#618]
In GitLab by @rainman110 on Nov 22, 2023, 08:56

Modules should save the GTlab core version (and possible the version of dependent modules) in its metadata.

I believe the current system version dependency management does not correctly work since
 - GTlab cannot figure out, if a module is compatible or not. It needs this inside the modules' meta data
 - A module defines its dependencies by version. This is however buggy, since this will never be updated. leading to a buggy behaviour.

Instead:
 - Each module needs to define a dependency rule (and the core as well). As an example, the GTlab core knows, that there are now ABI changes from 2.0 to 2.5 so it can safely import a module compiled for 2.1. So core needs to define a rule, which version range is okay to load.
 - To make this work, we need to compile in the core and dependent module versions into the metadata section. This is tricky, since we currently  can only extract this information from CMake.
 - The version of each module should be located also in the metadata section.
 - The modules's json file needs to be updated by cmake. We can do this using the configure_file command. This will place the actual version of all dependent components into the modules metadata

**Comment by jensschmeink** [comment]
@rainman110 The issue is older than 2 years. Is it still relevant?

**Comment by rainman110** [comment]
Yes. It is highly relevant! GTlab does not really know currently, what objects belong to which module, making it very hard to predict, what modules are actually required for a project.
